backend/src/bundles/auth/auth.service.ts

```diff
@@ -1,28 +1,85 @@
 import {
+    type UserSignInRequestDto,
+    type UserSignInResponseDto,
     type UserSignUpRequestDto,
     type UserSignUpResponseDto,
 } from '~/bundles/users/types/types.js';
+import { type UserRepository } from '~/bundles/users/user.repository.js';
 import { type UserService } from '~/bundles/users/user.service.js';
+import { cryptCompare } from '~/common/helpers/helpers.js';
+import { HttpCode, HttpError } from '~/common/http/http.js';
 import { tokenService } from '~/common/services/services.js';
 
 class AuthService {
     private userService: UserService;
+    private userRepository: UserRepository;
 
-    public constructor(userService: UserService) {
+    public constructor(
+        userService: UserService,
+        userRepository: UserRepository,
+    ) {
         this.userService = userService;
+        this.userRepository = userRepository;
     }
 
-    public async signIn(id: number): Promise<{ token: string }> {
+    public async signIn(id: number): Promise<UserSignInResponseDto> {
+        const user = await this.userService.findById(id);
+
         return {
+            ...user,
             token: await tokenService.create({ id }),
         };
     }
 
     public async signUp(
         userRequestDto: UserSignUpRequestDto,
     ): Promise<UserSignUpResponseDto> {
+        const { email } = userRequestDto;
+
+        const userByEmail = await this.userRepository.findByEmail(email);
+        if (userByEmail) {
+            throw new HttpError({
+                message: 'User already exist.',
+                cause: 'Email already used',
+                status: HttpCode.CONFLICT,
+            });
+        }
+
         const user = await this.userService.create(userRequestDto);
-        return this.signIn(user.toObject().id);
+        const { id } = user;
+
+        return {
+            ...user,
+            token: await tokenService.create({ id }),
+        };
+    }
+
+    public async verifyLoginCredentials({
+        email,
+        password,
+    }: UserSignInRequestDto): Promise<{ id: number; email: string }> {
+        const user = await this.userRepository.findByEmail(email);
+
+        if (!user) {
+            throw new Error('Incorrect email');
+        }
+
+        const isEqualPassword = await cryptCompare(password, password);
+        if (!isEqualPassword) {
+            throw new Error('Password not match');
+        }
+
+        return user.toObject();
+    }
+
+    public async login({ id }: { id: number }): Promise<UserSignInResponseDto> {
+        const user = await this.userService.findById(id);
+
+        return {
+            token: 'TOKEN', // Change with JWT
```

Comment:
you should return token

Reply:
This is the sign-in flow from another branch. And I'll fix it when the PR for sign-in is merged.